code/game/machinery/medical_pod/medical_pod.dm

```diff
@@ -0,0 +1,236 @@
+		return
+
+	go_out()
+	add_fingerprint(usr)
+	return
+
+/obj/structure/machinery/medical_pod/proc/extra_eject_checks()
+	return TRUE
+
+/obj/structure/machinery/medical_pod/MouseDrop_T(mob/target, mob/user)
+	. = ..()
+	var/mob/living/H = user
+	if(!istype(H) || target != user) //cant make others get in. grab-click for this
+		return
+
+	move_inside(target)
+
+/// the putter is the guy putting the person in the pod
+/obj/structure/machinery/medical_pod/proc/go_in(mob/M, mob/putter)
+	if(isXeno(M))
+		return
+
+	/// who is doing the work of putting/going in the scanner
+	var/main_character
+	if(!putter)
+		main_character = M
+	else
+		main_character = putter
+
+
+	if(go_in_timer)
+		if(!do_after(main_character, go_in_timer, INTERRUPT_NO_NEEDHAND, BUSY_ICON_GENERIC))
+			return
+
+	if(occupant)
+		to_chat(main_character, SPAN_BOLDNOTICE("\The [src] is already occupied!"))
+		return
+	if(main_character != M)
+		to_chat(main_character, SPAN_NOTICE("You move [M.name] inside \the [src]."))
+	else
+		to_chat(main_character, SPAN_NOTICE("You move into \the [src]."))
+
+	M.forceMove(src)
+	occupant = M
+	update_use_power(USE_POWER_ACTIVE)
+	update_icon()
+	//prevents occupant's belonging from landing inside the machine
+	for(var/obj/O in src)
+		O.forceMove(loc)
+
+/obj/structure/machinery/medical_pod/proc/go_out()
+	if(!(src.occupant))
+		return
+	for(var/obj/O in src)
+		O.forceMove(loc)
+
+	occupant.forceMove(loc)
+	occupant.update_med_icon()
+
+	if(exit_stun)
+		occupant.stunned = max(occupant.stunned, exit_stun) //Action delay when going out
+		occupant.update_canmove() //Force the delay to go in action immediately
+		occupant.visible_message(SPAN_WARNING("[occupant] pops out of \the [src]!"),
+		SPAN_WARNING("You get out of \the [src] and get your bearings!"))
+
+	occupant = null
+	update_use_power(USE_POWER_IDLE)
+	update_icon()
+	playsound(src, 'sound/machines/hydraulics_3.ogg')
+
+// attackby code
+
+/obj/structure/machinery/medical_pod/attackby(obj/item/W, mob/user)
+
+	if(!ishuman(user))
+		return // no
+	if(inoperable())
+		to_chat(user, SPAN_NOTICE("\The [src] is non-functional!"))
+		return
+	if(istype(W, /obj/item/grab))
+		var/mob/to_put_in
+		var/obj/item/grab/G = W
+		if(istype(G.grabbed_thing,/obj/structure/closet/bodybag/cryobag))
+			var/obj/structure/closet/bodybag/cryobag/C = G.grabbed_thing
+			if(!C.stasis_mob)
+				to_chat(user, SPAN_WARNING("\The [C] is empty!"))
+				return
+			to_put_in = C.stasis_mob
+			C.open()
+			user.start_pulling(to_put_in)
+		else if(ismob(G.grabbed_thing))
+			to_put_in = G.grabbed_thing
+		else
+			return
+		if(skilllock)
+			if(!skillcheck(usr, SKILL_SURGERY, skilllock))
+				to_chat(usr, SPAN_WARNING("You don't have the training to use \the [src]!"))
+				return
+		if(occupant)
+			to_chat(user, SPAN_WARNING("\The [src] is already occupied!"))
+			return
+		if(to_put_in.abiotic())
+			to_chat(user, SPAN_WARNING("Subject cannot have abiotic items on."))
+			return
+		if(isXeno(to_put_in))
```

Comment:
feel like this should be an ishumanstrict check right? why would you be able to put in jones or a predator

Reply:
just did ishuman because preds funny